core/spec/models/spree/payment_spec.rb

```diff
@@ -17,12 +17,12 @@
   let(:card) { create :credit_card }
 
   let(:payment) do
-    payment = Spree::Payment.new
-    payment.source = card
-    payment.order = order
-    payment.payment_method = gateway
-    payment.amount = 5
-    payment
+    Spree::Payment.create! do |payment|
```

Comment:
I had the same doubt and I noticed that in this same file payments are also created using [Spree::PaymentCreate.new](https://github.com/jhawthorn/solidus/blob/bb561a2327fbc6fc93af563c6154bbba08ea087e/core/spec/models/spree/payment_spec.rb#L693-L696) Service Object. Is there any reason/general rule to prefer a method or another?